Cloud gateways: How to upgrade data plane nodes (#7155)
Kong/docs.konghq.com: app/konnect/gateway-manager/data-plane-nodes/upgrade.md

Reviewer: Small nitpick that I don't think should be a blocker to merging this, but the first sentence in the first paragraph: "You can upgrade data plane nodes to a new {{site.base_gateway}} version by bringing
up new data plane nodes, and then shutting down the old ones." I think that only applies to self-managed nodes, so we might want to rewrite that. I think it's valuable to mention that it works by bringing up the new nodes and shutting down the old ones, but it's currently written in a way that makes it seem like you have to do that yourself.

Read through the [{{site.base_gateway}} upgrade considerations](/gateway/latest/upgrade/) for the version that you're upgrading to.

## Upgrade a data plane node
To upgrade a data plane node to a new version, follow these steps:

{% navtabs %}
{% navtab Dedicated Cloud Gateways %}
{% navtabs %}
{% navtab Konnect UI %}

> 1. Open [**Gateway Manager**](https://cloud.konghq.com/us/gateway-manager/), choose the Dedicated Cloud Gateways powered control plane,

Reviewer: ```suggestion
1. In [**Gateway Manager**](https://cloud.konghq.com/us/gateway-manager/), select the Dedicated Cloud Gateways powered control plane you want to upgrade the data plane nodes for.
```

> 1. From the **Actions** drop-down menu, select **Update Cluster Config**

Reviewer: ```suggestion
1. From the **Actions** menu, select **Update Cluster Config**.
```
Feel free to ignore, this just aligns us with Google's style guide recommendations.

> 1. Select the desired {{site.base_gateway}} version, and **Update Cluster**.

Reviewer: ```suggestion
1. Select the desired {{site.base_gateway}} version, and click **Update Cluster**.
```

> {{site.konnect_short_name}} will upgrade the version of all the data plane nodes to the version you selected automatically. 

Reviewer: ```suggestion
{{site.konnect_short_name}} will upgrade the version of all the data plane nodes to the version you selected automatically. 

Kong performs a rolling upgrade of the fully-managed data plane nodes. This is a zero downtime upgrade because Kong synchronizes the data plane with load balancer registration and deregistration and gracefully terminates the the old data plane nodes to reduce the impact on the ongoing traffic.
```
Something like this (edited from the eng response yesterday) to explain how the upgrade happens and what Kong does to make it zero downtime.

{% endnavtab %}
{% navtab API %}

Using the API, you can upgrade your data plane node versions by issuing a POST request to the `/cloud-gateways/configurations` endpoint: 

> 1. Open [**Gateway Manager**](https://cloud.konghq.com/us/gateway-manager/), choose the Dedicated Cloud Gateways powered control plane, and copy the `control_plane_id`.

Reviewer: ```suggestion
1. In [**Gateway Manager**](https://cloud.konghq.com/us/gateway-manager/), select the Dedicated Cloud Gateways powered control plane and copy the `control_plane_id`.
```
Would it be better to use the https://docs.konghq.com/konnect/api/control-planes/latest/#/Control%20Planes/list-control-planes endpoint to retrieve the control plane ID from the API? Just so that all of the steps stay in the API.

> 1. From the {{site.konnect_short_name}} sidebar, click on **Data Plane Nodes**, select the relevant data plane group, and record the `cloud_gateway_network_id`.

Reviewer: Could we use GET /cloud-gateways/networks for this?

1. Construct and send a POST request to `/cloud-gateways/configurations`, using both the `control_plane_id`, `cloud_gateway_network_id`, and the desired [`version`](/konnect/compatibility/):

    ```bash
    curl --request PUT \

>     --url https://us.api.konghq.com/v2/cloud-gateways/configurations \

Reviewer: ```suggestion
    --url https://<region>.api.konghq.com/v2/cloud-gateways/configurations \
```
I think that's the right notation?

>     --header 'Authorization: Bearer YOUR_API_KEY' \

Reviewer 2: question: Is this really an API key, or is a bearer token (JWT, PAT, or SPAT), as the authentication scheme implies?

Author: It's a bearer token

Author: I've pushed up a fix.

    --data '{
    "control_plane_id": "CONTROL_PLANE_ID",
    "version": "DESIRED_VERSION",
        "control_plane_geo": "GEO_LOCATION",
    "dataplane_groups": [
        {
        "provider": "aws",

>         "region": "ap-northeast-1",

Reviewer: Should this one also be a variable?

        "cloud_gateway_network_id": "CLOUD_GATEWAY_NETWORK_ID",
        "autoscale": {
                "kind": "autopilot",
                            "base_rps": 100
            }
        }
    ]
    }'
    ```

{:.note}
> Replace `BEARER_TOKEN`, `CONTROL_PLANE_ID`, `DESIRED_VERSION`, `GEO_LOCATION`, and `CLOUD_GATEWAY_NETWORK_ID` with your specific details.

> {{site.konnect_short_name}} will upgrade the version of all the data plane nodes to the version you selected automatically, the response body will contain information about the data plane group, including the version. 

Reviewer: ```suggestion
{{site.konnect_short_name}} will upgrade the version of all the data plane nodes to the version you selected automatically, the response body will contain information about the data plane group, including the version. 

Kong performs a rolling upgrade of the fully-managed data plane nodes. This is a zero downtime upgrade because Kong synchronizes the data plane with load balancer registration and deregistration and gracefully terminates the the old data plane nodes to reduce the impact on the ongoing traffic.
```
